cms/users/models.py

```diff
@@ -1,13 +1,82 @@
-from django.contrib.auth.models import AbstractUser
+import uuid
+from collections.abc import Iterable
+
+from django.conf import settings
+from django.contrib.auth.models import AbstractUser, Group
 from django.db import models
 from django.utils.functional import cached_property
 
+from cms.teams.models import Team
+
 
 class User(AbstractUser):
     """Barebones custom user model."""
 
+    # Unique identifier for user lookup, as emails (used as usernames) may change.
+    user_id = models.UUIDField(unique=True, default=uuid.uuid4, editable=False)
```

Comment:
Question: Why does this need to be different to the user's primary key? Having 2 different values for a user's "id" is likely to get confusing, especially for auditing.

Reply:
Thanks, Dan, updated!